plugin/pkg/scheduler/core/generic_scheduler.go

```diff
@@ -474,6 +475,10 @@ func podFitsOnNode(
 				if !fit {
 					// eCache is available and valid, and predicates result is unfit, record the fail reasons
 					failedPredicates = append(failedPredicates, reasons...)
+					// if alwaysCheckAllPredicates is false, short circuit all predicates when one predicate fails.
+					if !alwaysCheckAllPredicates {
+						break
```

Comment:
It would be better, if we log saying that `since alwaysCheckAllPredicates has been set, the predicate evaluation is short circuited and there are chances of other predicates failing as well`

Reply:
That makes sense.